backend/FwLite/MiniLcm/SyncHelpers/EntrySync.cs

```diff
@@ -162,27 +171,88 @@ public override Task<int> Replace(ComplexFormComponent beforeComponent, ComplexF
         }
     }
 
-    private class SensesDiffApi(IMiniLcmApi api, Guid entryId) : IOrderableCollectionDiffApi<Sense>
+    private class ComplexFormComponentsDiffApi(Entry afterEntry, IMiniLcmApi api) : IOrderableCollectionDiffApi<ComplexFormComponent>
+    {
+        private readonly bool supportsEntityIds = api.GetDataFormat() == ProjectDataFormat.Harmony;
```

Comment:
I really don't like this (and the code which is conditional on this). What if we used Id everywhere, and the Fw version set the component Id to the right thing based on context? eg: either the ComponentSense??EntryId or ComplexFormEntryId.